### PR TITLE
Image file size should not be issue anymore + other bug changes; check comments below.

### PR DESCRIPTION
**Pull request by manningwu07**
Fix https://github.com/AP-Students/website/issues/118

### CODE REVIEW

**Comment by manningwu07** [comment]
Branch conflicts are just moved files. You can just replace main branch w this branch.

**Comment by manningwu07** [comment]
Found major errors brb

**Comment by manningwu07** [comment]
ok nvm its fixed lmao i freaked out too much

**Comment by manningwu07** [comment]
Bug fixes: 
- file structure changes
- fixed "add chapter" state sharing issue
- Fixed subject sidebar not updating properly
- fixed some link issues with breadcrumb and table of contents to make them more robust.

File: src/app/admin/subject/[slug]/page.tsx [hunk 1]
```diff
@@ -14,11 +14,11 @@ import { Input } from "@/components/ui/input";
 import { useUser } from "@/components/hooks/UserContext";
 import { db } from "@/lib/firebase";
 import { doc, getDoc, setDoc } from "firebase/firestore";
-import { Link } from "../_components/link";
-import apClassesData from "@/app/admin/apClasses.json";
+import Link from "next/link";
```
Comment (BrilliantDeviation7) [inline src/app/admin/subject/[slug]/page.tsx:17]: This needs to be from the custom `link.tsx` component for the navigation blocker to work on Link components.

File: src/components/article-creator/ArticleCreator.tsx [hunk 2]
```diff
@@ -281,11 +332,19 @@ function ArticleCreator({ className }: { className?: string }) {
...
-            const updatedTable = await processTable(block.data);
+            const updatedTable = await processTable(block.data as TableData);
             block.data.content = updatedTable;
             return block;
           }
 
+          // Process images
+          if (block.type === "image" && block.data.url.startsWith("data:image/")) {
+            console.log("Blockdata", block.data);
+            const updatedImage = await processImage(block.data as ImageData);
```
Comment (BrilliantDeviation7) [inline src/components/article-creator/ArticleCreator.tsx:343]: Is this a better solution than https://github.com/editor-js/image?

**Review by BrilliantDeviation7 — CHANGES_REQUESTED** [review]
`link.tsx` and `navigation-block.tsx` should be moved to within `app/admin/subject` because they don't belong with the other user-facing subject page components (could be `app/admin/subject/_components`. Also see comments.

File: src/components/article-creator/ArticleCreator.tsx (hunk 2 above)
Comment (manningwu07) [reply src/components/article-creator/ArticleCreator.tsx:343]: that requires yarn and were running npm

Comment (manningwu07) [reply src/components/article-creator/ArticleCreator.tsx:343]: also it works well so like im fine w it

File: src/app/admin/subject/[slug]/page.tsx (hunk 1 above)
Comment (manningwu07) [reply src/app/admin/subject/[slug]/page.tsx:17]: Ok so just replace Link from "next/link" to { Link }?

**Review by BrilliantDeviation7 — APPROVED** [review]

**Pull request merged**